tests/e2e/operator/region.go

```diff
@@ -0,0 +1,612 @@
+}
+
+// VerifyHelmUpgrade waits till all the pods are restarted after the
+// helm upgrade is completed, it verifies with initialTimestamp which is the timestamp
+// of the pods before recreation and returns the pod name.
+func (r *Region) VerifyHelmUpgrade(t *testing.T, initialTimestamp time.Time, kubectlOptions *k8s.KubectlOptions) error {
+	// Wait for the pods to be recreated with a new timestamp after the upgrade.
+	_, err := retry.DoWithRetryE(t, "waiting for pods to be recreated with new timestamp",
+		60, 10*time.Second,
+		func() (string, error) {
+			// List the pods for the deployment.
+			pods := k8s.ListPods(t, kubectlOptions, metav1.ListOptions{
+				LabelSelector: LabelSelector,
+			})
+
+			// Check if any pods exist.
+			if len(pods) == 0 {
+				return "", fmt.Errorf("no pods found for deployment")
+			}
+
+			// Check if any pod has a creation timestamp greater than the initial timestamp.
+			// We are actually waiting for all pods to complete helm upgrade,
+			// as just verifying one pod and cleaning up the resources will cause issues,
+			// Since helm delete is happening while pods are still in upgrade process.
+			for _, pod := range pods {
+				if !pod.CreationTimestamp.Time.After(initialTimestamp) {
+					return "", fmt.Errorf("pod %s has not been recreated with a new timestamp yet", pod.Name)
+				}
+			}
+			return "", nil
+		})
+	return err
+}
+
+// ValidateMultiRegionSetup validates the multi-region setup.
+func (r *Region) ValidateMultiRegionSetup(t *testing.T) {
+	// Validate multi-region setup.
+	for _, cluster := range Clusters {
+		//Get current context name.
+		// Get current context name.
+		kubeConfig, rawConfig := r.GetCurrentContext(t)
+		rawConfig.CurrentContext = cluster
+		kubectlOptions := k8s.NewKubectlOptions(cluster, kubeConfig, r.Namespace[cluster])
+
+		pods := k8s.ListPods(t, kubectlOptions, metav1.ListOptions{
+			LabelSelector: LabelSelector,
+		})
+		require.NotEmpty(t, pods)
+
+		// Execute SQL query to verify regions
+		stdout, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions,
+			"exec", pods[0].Name, "-c", "cockroachdb", "--",
+			"/cockroach/cockroach", "sql", "--certs-dir=/cockroach/cockroach-certs",
+			"-e", "SHOW REGIONS FROM CLUSTER")
+		require.NoError(t, err)
+
+		// Verify regions output
+		expectedRegions := []string{
+			"k3d-us-east1",
+			"k3d-us-east2",
+		}
+		for _, clusterRegion := range expectedRegions {
+			require.Contains(t, stdout, clusterRegion)
+		}
+
+		// Execute node status command and verify node properties
+		nodeStatus, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions,
+			"exec", pods[0].Name, "-c", "cockroachdb", "--",
+			"/cockroach/cockroach", "node", "status", "--certs-dir=/cockroach/cockroach-certs")
+		require.NoError(t, err)
+
+		for _, clusterRegion := range expectedRegions {
+			require.Contains(t, nodeStatus, clusterRegion)
+		}
+		require.NotContains(t, nodeStatus, "false")
```

Comment:
Can we be more precise with this assertion? For example, we could assert that each region has the expected number of nodes, and each node is both live and available.

Reply:
I think the output will be easier to parse if we use `cockroach node status --format json`.